compute/cker/include/cker/train/operation/Loss.h

```diff
@@ -54,6 +54,23 @@ inline void MSE(const Shape &y_pred_shape, const T *y_pred_data, const Shape &y_
   output_data[0] = (T)(squared_sum / size);
 }
 
+template <typename T>
+inline void MSEGrad(const Shape &y_pred_shape, const T *y_pred_data, const Shape &y_true_shape,
+                    const T *y_true_data, const Shape &grad_shape, T *grad_data)
+{
+  if (y_pred_shape != y_true_shape)
+    throw std::runtime_error("cker::MSEGrad: y_pred_shape != y_true_shape");
+  if (y_pred_shape != grad_shape)
+    throw std::runtime_error("cker::MSEGrad: y_pred_shape != grad_shape");
+
+  const auto y_pred = MapAsMatrixWithLastDimAsRows(y_pred_data, y_pred_shape);
+  const auto y_true = MapAsMatrixWithLastDimAsRows(y_true_data, y_true_shape);
+  auto grad = MapAsMatrixWithLastDimAsRows(grad_data, grad_shape);
+
+  auto size = y_pred.cols() * y_pred.rows();
```

Comment:
If you want the number of all elements, you can use the code below.
```suggestion
  auto size = y_pred_shape.FlatSize();
```

Anyway I have a question.
Isn't it possible that the first dimension of  `y_pred` is batch size? If `y_pred` has a batch, shouldn't we calculate average per batch?

Reply:
As far as I know, the loss value is the sum or the mean over the batch. However, if user sets a Reduction option(?), user can get the loss of each batch size.
First of all, I didn't consider the Reduction case. How about considering it later? I will make another Grad function for this.

Reply:
> Type of [tf.keras.losses.Reduction](https://www.tensorflow.org/api_docs/python/tf/keras/losses/Reduction) to apply to loss. Default value is AUTO. AUTO indicates that the reduction option will be determined by the usage context. For almost all cases this defaults to SUM_OVER_BATCH_SIZE.
> 
> from https://www.tensorflow.org/api_docs/python/tf/keras/losses/MeanSquaredError

Yes, you're right. We have to consider how to support other `Reduction` (I don't know whether we use the same name as tensorflow) options later. :)

Reply:
I didn't mean that we use the same name as tensorflow. :)